### DISCUSSION TITLE
WPF上面显示海量图片的布局和控件的问题

### DISCUSSION THREAD

**Discussion by ivei** [BYJRK/DotNet-Discussions#161] [WPF 开发]
遇到一个问题: 
我有大约2000张需要拼接的照片, 需要拼接成一张完整的视野场景. 每张照片是1千万像素, 需要在界面上显示拼接后的效果.
已经将图片做好了金字塔效果, 生成了分层压缩的图像. 

以前是用Qt做的, Qt有GraphicsFrameWork, 这个事情很容易做, 因为QGraphicsScene天生支持动态按需加载, 很容易控制. 根据缩放比例, 选择合适的图层, 然后QGraphicsView自身在QGraphicsPixmapItem需要显示时会有请求, 我只要重载并load图像, 并释放不再显示的图片, 就可以将内存占用控制在最多4张图片的范围内. 

现在需要将软件转到WPF上面. 一种最笨的做法是用一个Canvas开始做起, 实时计算位置, 加载卸载图片. 但是这么做很笨重, 几乎等于把QGraphicsView/QGraphicsScene重写了一遍, 还不如人家功能强性能好. 

想问一下有什么其他的方法, 或者控件? 我一直没有在WPF上找到类似的东西

**Comment by designerlast** [comment]
可以使用graphic去drawimage到一个新的照片上，显示可以使用writebitmap支持后台缓存更新
